controllers/nstemplateset/space_roles.go

```diff
@@ -69,7 +69,7 @@ func (r *spaceRolesManager) ensure(ctx context.Context, nsTmplSet *toolchainv1al
 			return false, r.wrapErrorWithStatusUpdate(lctx, nsTmplSet, r.setStatusNamespaceProvisionFailed, err, "failed to provision namespace '%s' with space roles", ns.Name)
 		}
 
-		if err := deleteObsoleteObjects(lctx, r.Client, lastAppliedSpaceRoleObjs, spaceRoleObjs); err != nil {
+		if err := deleteObsoleteObjects(lctx, r.Client, lastAppliedSpaceRoleObjs, spaceRoleObjs, nsTmplSet); err != nil {
 			return false, r.wrapErrorWithStatusUpdate(lctx, nsTmplSet, r.setStatusUpdateFailed, err, "failed to delete redundant objects in namespace '%s'", ns.Name)
 		}
```

Comment:
As I see, you don't support feature-enabled resources for SpaceRole objects at all - you don't filter them during creation. So, do you really still want to keep the support of the namespace-scoped resources when it's so limited?

Reply:
I deleted all changes for namespaced resources. The PR supports the cluster scoped resource only now.